Recipes.Test/JsonConfiguration/JsonConfigurationTest.cs

```diff
@@ -238,5 +455,140 @@ private class InheritedType3 : IBaseInterface
 
             public string String { get; set; }
         }
+
+        [Bindable(true)]
+        private class Animal
+        {
+            protected Animal(int age, string name)
+            {
+                this.Age = age;
+                this.Name = name;
+            }
+
+            public string Name;
+
+            public int Age { get; }
+        }
+
+        private enum FurColor
+        {
+            Black,
+
+            Brindle,
+
+            Golden
+        }
+
+        private class Dog : Animal
+        {
+            public Dog(int age, string name, FurColor furColor)
+                : base(age, name)
+            {
+                this.FurColor = furColor;
+                this.DogTag = "my name is " + name;
+            }
+
+            public FurColor FurColor { get; }
+
+            public string DogTag { get; }
+        }
+
+        private class Cat : Animal
+        {
+            public Cat(int age, string name, int numberOfLives)
+                : base(age, name)
+            {
+                this.NumberOfLives = numberOfLives;
+            }
+
+            public int NumberOfLives;
+        }
+
+        private class Mouse : Animal
+        {
+            public Mouse(int age, string name, FurColor furColor, int tailLength)
+                : base(age, name)
+            {
+                this.FurColor = furColor;
+                this.TailLength = tailLength;
+            }
+
+            public FurColor FurColor { get; }
+
+            public int TailLength;
+        }
+
+        private class Tiger : Animal
+        {
+            public Tiger(int age, string name, int numberOfTeeth, int tailLength)
+                : base(age, name)
+            {
+                this.NumberOfTeeth = numberOfTeeth;
+                this.TailLength = tailLength;
+            }
+
+            public int NumberOfTeeth { get; }
+
+            public int TailLength;
+        }
+
+        [Bindable(true)]
+        private class Diet
+        {
+        }
+
+        private class Atkins : Diet
```

Comment:
I think we've already discussed on the phone.  Atkins is a well-known diet, and so its not parameterized by anything.  Whereas LowCalorie requires the number of calories.  And I need this type to test for deserializing empty json string {}